app/components/Views/confirmations/components/Confirm/StakingDetails/StakingDetails.test.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import renderWithProvider from '../../../../../../util/test/renderWithProvider';
+import { stakingDepositConfirmationState } from '../../../../../../util/test/confirm-data-helpers';
+import StakingDetails from './StakingDetails';
+import { decGWEIToHexWEI } from '../../../../../../util/conversions';
+import initialRootState from '../../../../../../util/test/initial-root-state';
+
+describe('TokenHero', () => {
+  it('contains token and fiat values for staking deposit', async () => {
+    const { getByText } = renderWithProvider(<StakingDetails />, {
+      state: {
+        ...stakingDepositConfirmationState,
+        engine: {
+          ...stakingDepositConfirmationState.engine,
+          backgroundState: {
+            ...stakingDepositConfirmationState.engine.backgroundState,
+            TransactionController: {
+              ...stakingDepositConfirmationState.engine.backgroundState.TransactionController,
+              transactions: [
+                {
+                  ...stakingDepositConfirmationState.engine.backgroundState.TransactionController.transactions[0],
+                  txParams: {
+                    ...stakingDepositConfirmationState.engine.backgroundState.TransactionController.transactions[0].txParams,
+                    value: `0x${decGWEIToHexWEI(10 ** 8)}`,
+                  },
+                }
+              ]
+            },
+          },
+        },
+        staking: {
+          ...initialRootState.staking,
+          vaultData: {
+            ...initialRootState.staking.vaultData,
+            apy: '2.2',
+          },
+        },
+      },
+    });
```

Comment:
```suggestion
const updatedState = merge({}, stakingDepositConfirmationState, {
      engine: {
        backgroundState: {
          TransactionController: {
            transactions: [
              {
                txParams: {
                  value: `0x${decGWEIToHexWEI(10 ** 8)}`,
                },
              },
            ],
          },
        },
      },
      staking: {
        vaultData: {
          apy: '2.2',
        },
      },
    });
    const { getByText } = renderWithProvider(<StakingDetails />, {
      state: updatedState,
    });
```
Could this be simpler with lodash merge?